unikernel.ml

```diff
@@ -1201,70 +1212,172 @@ struct
                                     ~api_meth:false `Internal_server_error reqd
                                     ()))))))
 
-  let process_unikernel_update ~unikernel_name ~job ~build
+  let force_create_unikernel ~unikernel_name ~unikernel_image
       (unikernel_cfg : Vmm_core.Unikernel.config) (user : User_model.user)
-      albatross http_client reqd =
-    Builder_web.send_request http_client
-      ("/job/" ^ job ^ "/build/" ^ build ^ "/main-binary")
+      albatross =
+    let unikernel_config = { unikernel_cfg with image = unikernel_image } in
+    Albatross.query albatross ~domain:user.name ~name:unikernel_name
+      (`Unikernel_cmd (`Unikernel_force_create unikernel_config))
     >>= function
-    | Error (`Msg err) ->
-        Logs.err (fun m ->
+    | Error err ->
+        Logs.warn (fun m ->
             m
-              "builds.robur.coop: Error while fetching the binary of %s with \
-               error: %s"
+              "albatross-force-create: error querying albatross for creating \
+               %s: %s"
               unikernel_name err);
-        Middleware.http_response reqd ~title:"Error"
+        Lwt.return
+          (Error (`Msg ("Force create: Error querying albatross: " ^ err)))
+    | Ok (_hdr, res) -> (
+        match Albatross_json.res res with
+        | Error (`String err) ->
+            Logs.warn (fun m ->
+                m
+                  "albatross-force-create: albatross_json.res error parsing \
+                   albatross response for creating %s: %s"
+                  unikernel_name err);
+            Lwt.return (Error (`Msg err))
+        | Ok res ->
+            Logs.info (fun m ->
+                m
+                  "albatross-force-create: succesfully created %s with result \
+                   %s"
+                  unikernel_name
+                  (Yojson.Basic.to_string res));
+            Lwt.return (Ok (Yojson.Basic.to_string res)))
+
+  let process_change ~unikernel_name ~job ~to_be_updated_unikernel
```

Comment:
I'm curious about the logic: shouldn't we first fetch the unikernel, check that the device matches -- before pushing that information that we updated the unikernel (and make a rollback possible)?

Reply:
I'm also curious: should we schedule it from being removed from the list after the 10 minutes?

Reply:
But maybe that's for later